R/genotype_plot.R

```diff
@@ -6,55 +27,37 @@ genotype_plot<-function(vcf=NULL,
                         cluster=FALSE,
                         snp_label_size=500000,
                         colour_scheme=c("#d4b9da","#e7298a","#980043")){
-  
-  # Usage:
-  # vcf = a bgzipped VCF 
-  # chr = chr of scaf ID
-  # start = start of region
-  # end = end of region
-  # gff = file with gene annotations ".gff3" format
-  # popmap = two column data frame with column 1 for individual IDs as they appear in the VCF and column 2 for pop labels
-  # cluster = whether to organise haplotypes by hclust clustering
-  # colour_scheme = character vector of colour values
-  
+
   # Note, the gff functionality is currently not used.
   
-  # Get what we need
-  lib<-c("ggdendro","vcfR","ggplot2","tidyr","data.table")
-  lapply(lib,library,character.only=T)
-  
   # Subset the VCF on the command-line for the chr of interest
-  if(substr(vcf, nchar(vcf)-2+1, nchar(vcf)) == "gz"){
-    system(paste0("bcftools view -r ",chr,":",start,"-",end," ",vcf," > gt_plot_tmp.vcf"),wait=TRUE)
+  vcf_tempfile <- tempfile(pattern = "gt_plot", fileext = '.vcf')
+  on.exit({ unlink(vcf_tempfile) })
+  if(tools::file_ext(vcf) == "gz"){
+    system(glue::glue("bcftools view -r {chr}:{start}-{end} {vcf} > {vcf_tempfile}"), wait=TRUE)
   } else {
     stop("VCF needs to be bgzipped")
   }
   
   # Tidy up the popmap and turn into a list
-  popmap2 <- lapply(unique(popmap[,2]),function(pop){
-    return(as.character(popmap[popmap[,2]==pop,1]))
-  })
-  names(popmap2)<-unique(popmap[,2])
+  popmap <- as.data.table(popmap)
+  popmap2 <- lapply(split(popmap[,1], popmap[,2]), unlist, use.names = FALSE)
```

Comment:
split here reorders populations so they're alphabetical in popmap2 as opposed to the order they appear in the unique(popmap[,2]), which is the original functionality. Would just need to add an additional line to reorder popmap2 so names(popmap2) == unique(popmap[,2]), or revert to the original unless there is another reason to make this change?